lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1312 @@
+    """
+    Returns a FHIR Encounter resource entry and reference for the encounter in the study.
+    """
+
+    def grab_symptom_key(key: str, variable_name: str) -> Optional[Match[str]]:
+        if record[key] == '1':
+            return re.match(f"{variable_name}___[a-z_]+", key)
+        else:
+            return None
+
+    def build_condition(symptom: str, onset_date: str) -> dict:
+        return create_resource_condition(record, symptom, patient_reference, onset_date)
+
+    def build_diagnosis(symptom: str) -> Optional[dict]:
+        mapped_symptom = map_symptom(symptom)
+        if not mapped_symptom:
+            return None
+
+        return { "condition": { "reference": f"#{mapped_symptom}" } }
+
+    def build_locations_list(location: dict) -> dict:
+        return {
+            "location": create_reference(
+                reference_type = "Location",
+                reference = location["fullUrl"]
+            )
+        }
+
+    def non_tract_locations(resource: dict):
+        return bool(resource) \
+            and resource['resource']['identifier'][0]['system'] != f"{INTERNAL_SYSTEM}/location/tract"
+
+    # Map the various symptoms variables to their onset date.
+    # For daily_symptoms_covid_like use the attestation_date - 24 hours because the question asks
+    # "in the past 24 hours"
+    if event_type == EventType.ENCOUNTER:
+        symptom_onset_map = {
+            'daily_symptoms_covid_like': (datetime.strptime(record['attestation_date'], '%Y-%m-%d') \
+                - timedelta(hours=24)).strftime('%Y-%m-%d'),
+            'symptoms': record['symptom_onset'],
+            'symptoms_kiosk': record['symptom_duration_kiosk'],
+            'symptoms_swabsend': record['symptom_duration_swabsend']
+        }
+    elif event_type == EventType.ENROLLMENT:
+        symptom_onset_map = {'symptoms_base': record['symptom_onset_base']}
+
+    contained = []
+    diagnosis = []
+
+    for symptom_variable in symptom_onset_map:
+        symptom_keys = []
+
+        for redcap_key in record.keys():
+            symptom_key = grab_symptom_key(redcap_key, symptom_variable)
+            if symptom_key:
+                symptom_keys.append(symptom_key.string)
+
+        symptoms = list(map(lambda x: re.sub('[a-z_]+___', '', x), symptom_keys))
+        for symptom in symptoms:
+            contained.append(build_condition(symptom, symptom_onset_map[symptom_variable]))
+            diagnosis.append(build_diagnosis(symptom))
+
+    encounter_identifier = create_identifier(
+        system = f"{INTERNAL_SYSTEM}/encounter",
+        value = f"{REDCAP_URL}{record.project.id}/{record['record_id']}/{record['redcap_repeat_instance']}"
+    )
+
+    collection_code = None
+
+    if event_type == EventType.ENROLLMENT or collection_method == CollectionMethod.SWAB_AND_SEND:
+        collection_code = "HH"
+    elif collection_method == CollectionMethod.KIOSK:
+        collection_code = "FLD"
```

Comment:
What do HH and FLD stand for here? HH=Household?

Reply:
Good question. I added a comment about those. HH = Household and FLD = Field